hibernate-core/src/main/java/org/hibernate/cfg/EJB3DTDEntityResolver.java

```diff
@@ -137,7 +175,51 @@ private InputSource buildInputSource(String publicId, String systemId, InputStre
 	private InputStream getStreamFromClasspath(String fileName) {
 		LOG.trace( "Recognized JPA ORM namespace; attempting to resolve on classpath under org/hibernate/jpa" );
 		String path = "org/hibernate/jpa/" + fileName;
-		InputStream dtdStream = resolveInHibernateNamespace( path );
+		InputStream dtdStream = ResourcesHelper.locateResourceAsStream( path, this.getClass().getClassLoader() );
 		return dtdStream;
 	}
+
+	private InputSource resolveOnClassPath(String publicId, String systemId, String namespace) {
+		InputSource source = null;
+		String path = "org/hibernate/" + systemId.substring( namespace.length() );
+		InputStream dtdStream = ResourcesHelper.locateResourceAsStream( path, this.getClass().getClassLoader() );
+		if ( dtdStream == null ) {
+			LOG.debugf( "Unable to locate [%s] on classpath", systemId );
+			if ( systemId.substring( namespace.length() ).indexOf( "2.0" ) > -1 ) {
+				LOG.usingOldDtd();
+			}
+		}
+		else {
+			LOG.debugf( "Located [%s] on classpath", systemId );
+			source = new InputSource( dtdStream );
+			source.setPublicId( publicId );
+			source.setSystemId( systemId );
+		}
+		return source;
+	}
+
+	// Copied over from ConfigHelper that was removed
+	private InputStream resolveInLocalNamespace(String path) {
+		boolean hasLeadingSlash = path.startsWith( "/" );
+		String stripped = hasLeadingSlash ? path.substring( 1 ) : path;
+
+		InputStream stream = null;
+
+		ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+		if ( classLoader != null ) {
+			stream = classLoader.getResourceAsStream( path );
+			if ( stream == null && hasLeadingSlash ) {
+				stream = classLoader.getResourceAsStream( stripped );
+			}
+		}
+
+		if ( stream == null ) {
+			stream = Environment.class.getClassLoader().getResourceAsStream( path );
+		}
+		if ( stream == null && hasLeadingSlash ) {
+			stream = Environment.class.getClassLoader().getResourceAsStream( stripped );
```

Comment:
Could we move the 'stripped' creation logic into this if block? We can enjoy two benefits:

- better code readability for the variable creation is near its usage;
- potential avoidance of unnecessary performance cost for obviously it is perfectly normal when we don't need to create the 'stripped' string.

I would even go as far as below:
```
if ( stream == null && path.startsWith( "/" ) {
    stream = Environment.class.getClassLoader().getResourceAsStream( path.substring( 1 ) );
}
```